FoodOrderingApp-service/src/main/java/com/upgrad/FoodOrderingApp/service/businness/OrderService.java

```diff
@@ -0,0 +1,149 @@
+package com.upgrad.FoodOrderingApp.service.businness;
+
+import com.upgrad.FoodOrderingApp.service.dao.*;
+import com.upgrad.FoodOrderingApp.service.entity.*;
+import com.upgrad.FoodOrderingApp.service.exception.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class OrderService {
+
+    @Autowired
+    private OrderDao orderDao;
+
+    @Autowired
+    private CouponDao couponDao;
+
+    @Autowired
+    private AddressDao addressDao;
+
+    @Autowired
+    private PaymentDao paymentDao;
+
+    @Autowired
+    private RestaurantDao restaurantDao;
+
+    @Autowired
+    private ItemDao itemDao;
+
+    @Autowired
+    AuthorizationService authorizationService;
+
+    @Autowired
+    private CustomerDao customerDao;
+
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public List<CouponEntity> getCouponByName(String authorization,String couponName) throws CouponNotFoundException, AuthorizationFailedException, AuthenticationFailedException {
+        CustomerAuthTokenEntity customerAuthTokenEntity = authorizationService.fetchAuthTokenEntity(authorization);
```

Comment:
Variable authorization will be of the form "Bearer accesstoken" please remove the word bearer, otherwise all authorization checks will keep failing